src/RestrictedPython/PrintCollector.py

```diff
@@ -12,37 +12,28 @@
 ##############################################################################
 from __future__ import print_function
 
-import sys
-
-
-version = sys.version_info
-
 
 class PrintCollector(object):
     """Collect written text, and return it when called."""
 
-    def __init__(self):
+    def __init__(self, _getattr_=None):
         self.txt = []
+        self._getattr_ = _getattr_
 
     def write(self, text):
         self.txt.append(text)
 
     def __call__(self):
         return ''.join(self.txt)
 
+    def _call_print(self, *objects, **kwargs):
+        if not 'file' in kwargs:
+            kwargs['file'] = self
 
-printed = PrintCollector()
-
+        elif kwargs['file'] is None:
+            kwargs['file'] = self
```

Comment:
I think it would be easier to use `if kwargs.get('file', None) is None:` instead of the if and elif statements.